src/Runner.php

```diff
@@ -178,17 +183,61 @@ public function run($input = null, $output = null, $app = null, $commandFiles =
         return $statusCode;
     }
 
+    public function setLoader(ClassLoader $loader)
+    {
+        return $this->setCommandFilePluginPrefixes($loader->getPrefixesPsr4());
+    }
+
+    public function setCommandFilePluginPrefixes($prefixes)
+    {
+        $this->commandFilePluginPrefixes = $prefixes;
+        return $this;
+    }
+
+    public function setCommandFilePluginPattern($namespacePattern)
+    {
+        $this->namespacePattern = $namespacePattern;
+        return $this;
+    }
+
+    protected function getPluginCommandClasses()
+    {
+        $commandClasses = [];
```

Comment:
Yes and no. I don't think it would be good for folks to be calling methods of the runner to do this, but we could provide a plugin discovery class that provided this service. Then Runner could use the new class.